plugin/trino-hive/src/main/java/io/trino/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -370,7 +370,10 @@ private void invokeNoMoreSplitsIfNecessary()
         TupleDomain<HiveColumnHandle> effectivePredicate = compactEffectivePredicate.transformKeys(HiveColumnHandle.class::cast);
 
         List<HiveColumnHandle> partitionColumns = getPartitionKeyColumnHandles(table, typeManager);
-        BooleanSupplier partitionMatchSupplier = () -> partitionMatches(partitionColumns, dynamicFilter.getCurrentPredicate(), hivePartition);
+        BooleanSupplier partitionMatchSupplier =
+                partitionColumns.stream().anyMatch(dynamicFilter.getColumnsCovered()::contains)
+                        ? () -> true
```

Comment:
I think this should the the other way around: if there is covering column then use `partitionMatches(partitionColumns, dynamicFilter.getCurrentPredicate(), hivePartition)`

Reply:
of course, you're correct.
this indicates some missing test coverage, since i just nuked a feature, and build is still green. 
@raunaqmorarka @sopel39 do you have idea how this could be tested? can you test cover that?

Reply:
There is `TestHiveDynamicPartitionPruningTest` but I guess it's not complete due to https://github.com/trinodb/trino/issues/5172?